fabric-datafixer-v1/src/main/java/net/fabricmc/fabric/impl/datafixer/FabricDataFixerImpl.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.datafixer;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+
+import com.google.common.base.Preconditions;
+import com.mojang.datafixers.DataFixer;
+import com.mojang.datafixers.Dynamic;
+
+import net.fabricmc.fabric.api.datafixer.v1.DataFixerUtils;
+import net.fabricmc.fabric.api.util.NbtType;
+import net.minecraft.datafixers.DataFixTypes;
+import net.minecraft.datafixers.NbtOps;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.Tag;
+
+public final class FabricDataFixerImpl implements DataFixerUtils {
+
+	private final Map<String, DataFixerEntry> modFixers = new HashMap<>();
+	private boolean locked;
+
+	private FabricDataFixerImpl() {
+	}
+
+	public static final FabricDataFixerImpl INSTANCE = new FabricDataFixerImpl();
+
+	public void addFixerVersions(CompoundTag compoundTag) {
+		for (Entry<String, DataFixerEntry> entry : modFixers.entrySet()) {
+			compoundTag.putInt(entry.getKey() + "_DataVersion", entry.getValue().runtimeDataVersion);
+		}
+	}
+
+	@Override
+	public int getModDataVersion(CompoundTag compoundTag, String modid) {
+		return compoundTag.containsKey(modid + "_DataVersion", NbtType.NUMBER) ? compoundTag.getInt(modid + "_DataVersion") : 0;
```

Comment:
`compoundTag.getInt` does exactly the same thing as this whole line.

Reply:
So getInt will return 0 if no element is present? If it doesn't return 0 then it is an issue since the default data version is 0

Reply:
yes it returns 0 for nonexistent tags